README.md

```diff
@@ -296,3 +296,147 @@ JSArray args = new JSArray();
 // ...
 connection.event("interfaceName", "methodName", args);
 ```
+
+### JSTP compiler
+
+JSTP compiler is a nice feature for better handling of JSTP packets. You can declare interfaces comfortable for concrete handling purposes and use them instead of `ManualHandler`.
+
+#### Installation
+
+[Check for the latest version](https://bintray.com/metarhia/maven/jstp-compiler)
+
+Gradle:
+```
+dependencies {
+  compile 'com.metarhia.jstp:jstp-compiler:0.1.12'
+}
+```
+
+Maven:
+
+```
+<dependency>
+  <groupId>com.metarhia.jstp</groupId>
+  <artifactId>jstp-compiler</artifactId>
+  <version>0.1.12</version>
+  <type>pom</type>
+</dependency>
+```
+
+#### Handlers usage
+
+You can create your own `JSTPHandler` like this:
+
+```java
+@JSTPHandler
+public interface ExampleHandler {
+  // ...
+}
+```
+You can declare your own methods that will be called in `onInvoke()` method of basic ManualHandler.
+
+##### Named
+
+Gets the field of the received packet by specified name. Can be multi-level.
+
+```java
+@JSTPHandler
+public interface ExampleHandler {
+  // ...
+  @NotNull
+  @Named("ok")
+  void onOK(JSArray args);
+
+  @NotNull
+  @Named("error")
+  void onError(JSArray args);
+
+  // gets value by key "neededValue" in object got by "ok"
+  @Named(value = {"ok", "neededValue"})
+  void onNeededValueGot(String value);
+  // ...
+}
+```
+
+##### NotNull
+
+Checks if the needed value is not null before calling the handler method.
+
+```java
+@JSTPHandler
+public interface ExampleHandler {
+  // ...
+  @NotNull
+  void onExampleValue(JSArray args);
+  // ...
+}
+```
+
+##### Indexed
+
+Can be used for getting concrete argument from the received arguments. Can be multi-level
```

Comment:
for getting -> to get. Or maybe it's better to change the phrase as it has too much 'argument' in it =).